core/types/transaction.go

```diff
@@ -378,7 +378,7 @@ func (tx *Transaction) RollupCostData() RollupCostData {
 	if v := tx.rollupCostData.Load(); v != nil {
 		return v.(RollupCostData)
 	}
-	data, err := tx.MarshalBinary()
+	data, err := tx.WithoutBlobTxSidecar().MarshalBinary()
```

Comment:
I think we should add a few comments about why use non BLOB (but with BLOB hash) version of tx.

Another question is in which case the tx will contain the BLOB (as the derivation stage, we cannot derive the BLOB from L1).

Reply:
I've updated with some comments.